src/main/java/edu/harvard/iq/dataverse/DatasetPage.java

```diff
@@ -502,31 +487,46 @@ public boolean isSwiftStorage() {
     }
     
     //This function applies to a single datafile
-    public boolean isSwiftStorage(FileMetadata metadata){
+    private boolean isSwiftStorage(FileMetadata metadata){
         if (metadata.getDataFile().getStorageIdentifier().startsWith("swift://")) {
             return true;
         }
         return false;
     }
     
+    
+    private Boolean showComputeButtonForDataset = null;
     //This function applies to an entire dataset
     public boolean showComputeButton() {
+        if (showComputeButtonForDataset != null) {
+            return showComputeButtonForDataset;
+        }
+        
         if (isSwiftStorage() && (settingsService.getValueForKey(SettingsServiceBean.Key.ComputeBaseUrl) != null)) {
-            return true;
+            showComputeButtonForDataset = true;
+        } else {
+            showComputeButtonForDataset = false;
         }
-        return false;
+        return showComputeButtonForDataset;
     }
     
+    private Boolean showComputeButtonForFile = null; 
     //this function applies to a single datafile
     public boolean showComputeButton(FileMetadata metadata) {
+        if (showComputeButtonForFile != null) {
+            return showComputeButtonForFile;
+        }
+        
         if (isSwiftStorage(metadata) && (settingsService.getValueForKey(SettingsServiceBean.Key.ComputeBaseUrl) != null)) {
-            return true;
+            showComputeButtonForFile = true;
+        } else {
+            showComputeButtonForFile = false;
```

Comment:
same here

Reply:
OK, yeah - I screwed up with this one. The idea was to do the same thing as with the method above; to check it once, then cache the result. I just realized that this one is called on *different* files. So I can't just have one boolean to cache the result (doh). In order to cache these results I would need a Map of FileId-to-booleans. But the thing is, this probably doesn't need to be cached in the first place. Since this test is actually cheap.